waddrmgr/db.go

```diff
@@ -292,6 +376,64 @@ func fetchWatchingOnly(tx walletdb.Tx) (bool, error) {
 	return buf[0] != 0, nil
 }
 
+// fetchNumAllAddresses loads the metadata - number of all addrs - from the database.
+func fetchNumAllAddresses(tx walletdb.Tx) (uint32, error) {
```

Comment:
A uint32 is not sufficient to hold the potential number of addresses.  There can by 2^31 - 1 non-hardened addresses per account and up to 2^31 - 2 (one is reserved for imported addresses) accounts.  Thus, the max potential number of addresses is 2^62 - 2^32 - 2^31 + 2.  So a uint64 will cover it.